scout/ecm_prep.py

```diff
@@ -6447,6 +6515,64 @@ def fill_mkts(self, msegs, msegs_cpl, convert_data, tsv_data_init, opts,
         else:
             print("Success" + bstk_msg + bcpl_msg + bcc_msg + cc_msg)
 
+    def use_deflt_res_choice(self, mskeys, consume_warn, opts):
+        """Assign default res. tech. choice coefficients for segments without AEO coefficients.
+
+        Args:
+            mskeys (tuple): Current mseg information.
+            consume_warn (list): Previously warned-about segments w/ missing choice information.
+            opts (object): Stores user-specified execution options.
+
+        Returns:
+            Default choice coefficients for the end use and fuel type that are represented by
+            the segment with missing data.
+        """
+
+        # Set default coefficients to use in the case of missing data
+        if "electricity" in mskeys:
+            deflt_coefs = self.handyvars.deflt_res_choice["electric"][mskeys[4]]
+        else:
+            deflt_coefs = self.handyvars.deflt_res_choice["non-electric"][mskeys[4]]
+        # Record missing consumer data for primary technologies; if in verbose mode
+        # and the user has not already been warned about missing data for the given
+        # technology, print warning message
+        if mskeys[4] not in consume_warn:
+            consume_warn.append(mskeys[4])
+            verboseprint(opts.verbose, "WARNING: ECM '" + self.name + "' missing "
+                         "valid consumer choice data for segment '" + str(mskeys) +
+                         "'; using default choice data for refrigeration end use", "warning")
```

Comment:
Nice verbose warning message! Still wondering if error vs warning is better, which we can discuss later.